### DISCUSSION TITLE
https://angular.love/pl/angular-2-custom-form-controls

### DISCUSSION THREAD

**Discussion by giscus** [HouseOfAngular/angular-love#332] [General]
# https://angular.love/pl/angular-2-custom-form-controls

Angular.love - ciekawostki oraz rozwiązania dla średnio-zaawansowanych oraz zaawansowanych developerów Angulara.

https://angular.love/pl/angular-2-custom-form-controls

**Comment by roietik** [comment]
W szablonie html dla przycisku brakuje atrybutu type z wartością button.
Przycisk bez atrybutu type="button" może nieoczekiwanie wywołać ngSubmit formularza.
Jeśli nie podasz atrybutu type dla elementu <button>, przeglądarka domyślnie potraktuje go jako przycisk typu submit.
W Angular, dyrektywa ngSubmit jest związana z formularzami i służy do obsługiwania zdarzenia przesłania formularza. Gdy przycisk wywołuje przesłanie formularza (np. poprzez domyślne zachowanie type="submit"), Angular automatycznie wywoła funkcję podaną w atrybucie ngSubmit.
